templates/web/base/report/updates.html

```diff
@@ -1,5 +1,11 @@
 [% global.last_state = '' %]
+[% prev_state = '' %]
 [% FOREACH update IN updates %]
+[%- IF prev_state == 'hidden' OR prev_state == 'unconfirmed' OR update.problem_state == 'hidden' OR update.problem_state == 'unconfirmed' %]
```

Comment:
If (somehow, I guess it shouldn't but I wouldn't like to promise) a report has three updates, first has problem_state `hidden`, second has blank problem_state, third has problem_state `confirmed`, then this loop will show the third update that unhid the report, when that doesn't appear to be its intention.

It feels like you should be able to use `global.last_state` here, which should be doing exactly the same job as this `prev_state`. If the logic setting global.last_state was moved up here, then that would simplify it all I think?